punnetsquaremaker.py

```diff
@@ -15,7 +15,28 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 # The author can be reached by email at catherine.moresco@gmail.com.
+# Improved with Latex tables by Remyslab, email at remyslab@laposte.net.
 
+import datetime
+
+def headtex():
```

Comment:
Please consider refactoring so the file is only opened/closed once--opening and closing files is a relatively expensive operation. For example, you might want to make these functions return strings, and then write them to the file all at once in the main loop.

https://stackoverflow.com/questions/11349020/python-file-open-close-everytime-vs-keeping-it-open-until-the-process-is-finishe